File: mlflow-java/src/test/scala/com/databricks/mlflow/client/scala/BaseTest.scala
package com.databricks.mlflow.client.scala

import org.testng.Assert._
import org.testng.annotations._
import com.databricks.api.proto.mlflow.Service._
import com.databricks.mlflow.client.ApiClient
import scala.util.Properties

class BaseTest() {
  val apiUriDefault = "http://localhost:5001"
  var client: ApiClient = null 

  @BeforeClass
  def BeforeClass() {
      val apiUri = Properties.envOrElse("MLFLOW_TRACKING_URI",apiUriDefault)
      client = new ApiClient(apiUri)
  }

  def createExperimentName() = "TestScala_"+System.currentTimeMillis.toString
}

File: mlflow-java/src/test/java/com/databricks/mlflow/client/BaseTest.java
package com.databricks.mlflow.client;

import java.util.List;
import org.apache.log4j.Logger;
import org.testng.Assert;
import org.testng.annotations.*;

public class BaseTest {
    private static final Logger logger = Logger.getLogger(BaseTest.class);
    static String apiUriDefault = "http://localhost:5001";
    static ApiClient client ;

    @BeforeSuite
    public static void beforeSuite() throws Exception {
        logger.info("apiUriDefault="+apiUriDefault);
        String apiUriProp = System.getenv("MLFLOW_TRACKING_URI");
        logger.info("apiUriProp="+apiUriProp);
        String apiUri = apiUriProp == null || apiUriProp.length()==0 ? apiUriDefault : apiUriProp;
        logger.info("apiUri="+apiUri);
        client = new ApiClient(apiUri);
        client.setVerbose(true);
    }
}
